lib/transactions/transaction.ts

```diff
@@ -0,0 +1,228 @@
+import { CryptoUtil } from "../crypto-util.js";
+import { RemoteAPICaller } from "../gmd-api-caller.js";
+import { Signer } from "../signer.js";
+
+import Converters = CryptoUtil.Converters;
+
+export enum TransactionState {
+    ERROR = 0,

```

Comment:
The rule of thumb is enums should be better off as string enums. 

There's a whole debate around string vs number vs hybrid enums. Having it as string is the safest in case someone decides to add something new here without adding the actual value. The other thing to consider is now an enum is converted to javascript and string enums are the "best" ones overall. Hope you can make them a string.
```
export enum TransactionState {
    ERROR = 'error',
....
}
```

Reply:
Sounds good, thanks.